yrs/src/transaction.rs

```diff
@@ -602,21 +590,20 @@ impl Transaction {
             if let Some(mut list) = blocks.get(client) {
                 for r in range.iter() {
                     if let Some(pivot) = list.find_pivot(r.start) {
-                        let block = &list[pivot];
-                        if block.id().clock < r.start {
-                            blocks.split_block(&BlockPtr::new(
-                                ID::new(*client, r.start),
-                                pivot as u32,
-                            ));
+                        let block = list.get(pivot);
+                        let clock = block.id().clock;
+                        if clock < r.start {
+                            blocks.split_block(block, r.start - clock);
                             list = blocks.get(client).unwrap();
                         }
                     }
 
                     if let Some(pivot) = list.find_pivot(r.end) {
-                        let block = &list[pivot];
-                        if block.id().clock + block.len() > r.end {
-                            blocks
-                                .split_block(&BlockPtr::new(ID::new(*client, r.end), pivot as u32));
+                        let block = list.get(pivot);
+                        let block_id = block.id();
+                        let block_len = block.len();
+                        if block_id.clock + block_len > r.end {
+                            blocks.split_block(block, block_id.clock + block_len - r.end);
```

Comment:
Make sure that this method (and possibly others) use `get_item_clean_start` instead because it adds it to `merge_items` in Transaction.